flow/project.py

```diff
@@ -4065,11 +4008,19 @@ def hello(job):
         if isinstance(func, str):
             return lambda op: cls.operation(op, name=func)
 
+        if func in chain(
+            *cls._OPERATION_PRECONDITIONS.values(),
+            *cls._OPERATION_POSTCONDITIONS.values(),
+        ):
+            raise ValueError("A condition function cannot be used as an operation")
+
         if name is None:
             name = func.__name__
 
-        if (name, func) in cls._OPERATION_FUNCTIONS:
+        if name in map(lambda x: x[0], cls._OPERATION_FUNCTIONS):
             raise ValueError(f"An operation with name '{name}' is already registered.")
+        if func in map(lambda x: x[1], cls._OPERATION_FUNCTIONS):
+            raise ValueError("An operation with this function is already registered.")
```

Comment:
The structure of `x` and meaning of `x[0]` are not clear from this pattern. We can avoid the map/lambda and use readable names. Here are two suggestions, pick whichever you prefer.
```suggestion
        if any(name == op_name for op_name, op_func in cls._OPERATION_FUNCTIONS):
            raise ValueError(f"An operation with name '{name}' is already registered.")
        if any(func == op_func for op_name, op_func in cls._OPERATION_FUNCTIONS):
            raise ValueError("An operation with this function is already registered.")
```
Or iterate just one time:
```suggestion
        for op_name, op_func in cls._OPERATION_FUNCTIONS:
            if name == op_name:
                raise ValueError(f"An operation with name '{name}' is already registered.")
            elif func == op_func:
                raise ValueError("An operation with this function is already registered.")
```